### PR TITLE
Add rule for CVE-2024-4577

### PR DESCRIPTION
**Pull request by egibs**
This PR is an attempt to detect the conditions for `CVE-2024-4577` --
- https://www.wiz.io/blog/critical-rce-php-cgi-vulnerability
- https://github.com/watchtowrlabs/CVE-2024-4577
- https://labs.watchtowr.com/no-way-php-strikes-again-cve-2024-4577
- https://pentesterlab.com/exercises/cve-2012-1823

This will match the specific POC request which matches this from the `pentesterlab` post for `CVE-2012-1823`:
CleanShot 2024-06-10 at 14 38 34@2x

`allow_url_include` and `auto_prepend_file` were covered in the Dodgy PHP rules (separately), but I didn't see a soft-hyphen rule unless searching failed me.

### CODE REVIEW

File: rules/combo/backdoor/php.yara [hunk 1]
```diff
@@ -277,3 +277,20 @@ rule php_str_replace_obfuscation : critical {
...
+    date = "2024-06-06"
+    description = "detect php CGI argument injections"
+    discovered_by = "Orange Tsai (@orange_8361) of DEVCORE (@d3vc0r3)"
+    exploiters = "Aliz (@AlizTheHax0r) and Sina Kheirkhah (@SinSinology) of watchTowr (@watchtowrcyber)"
+    reference = "https://devco.re/blog/2024/06/06/security-alert-cve-2024-4577-php-cgi-argument-injection-vulnerability-en"
+    repository = "https://github.com/watchtowrlabs/CVE-2024-4577"
+    technical = "https://labs.watchtowr.com/no-way-php-strikes-again-cve-2024-4577"
+
+  strings:
+    $url_pattern = /\?%ADd\+allow_url_include%3d1\+(%ADd|-)[d+]\+auto_prepend_file%3dphp:\/\/input/ nocase
```
Comment (tstromberg) [inline rules/combo/backdoor/php.yara:292]: - I think you can leave out `nocase` here. See https://github.com/Neo23x0/YARA-Performance-Guidelines/ for why it should be avoided when possible.
- Removing the leading `\?%ADd\+` portion may yield more hits without adding false positives

File: rules/combo/backdoor/php.yara [hunk 2]
```diff
@@ -277,3 +277,20 @@ rule php_str_replace_obfuscation : critical {
 	condition:
 		filesize < 65535 and 2 of ($f*) and any of ($i*) and 2 of ($o*)
 }
+
+rule php_cgi_argument_injection : critical {
```
Comment (tstromberg) [inline rules/combo/backdoor/php.yara:281]: Technically this isn't a "combo" rule to match a backdoor.

What do you think about a new directory for these kinds of rules, for example: `exploits/php`?

**Review by tstromberg — CHANGES_REQUESTED** [review]

File: rules/combo/backdoor/php.yara (hunk 2 above)
Comment (egibs) [reply rules/combo/backdoor/php.yara:281]: I think that makes sense -- TBH the categorizations are still something I'm getting up to speed on 😅

Comment (egibs) [reply rules/combo/backdoor/php.yara]: Moved in [`8eb5607` (#266)](https://github.com/chainguard-dev/bincapz/pull/266/commits/8eb56076badf207af55fabb41c89a4377dc438e6).

File: rules/combo/backdoor/php.yara (hunk 1 above)
Comment (egibs) [reply rules/combo/backdoor/php.yara]: Removed `nocase` in [`f17f84d` (#266)](https://github.com/chainguard-dev/bincapz/pull/266/commits/f17f84d67fb09ae9e294b6b4fa020cc9c4cc5240), though 👍🏻

**Review by tstromberg — APPROVED** [review]

**Pull request merged**